src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -795,5 +775,67 @@
       },
       tableWidth: 100
     }
+  ],
+  listeners: [
+    {
+      name: 'approveWithMemo',
+      code: function(memo) {
+        var approvedApprovalRequest = this.clone();
+        approvedApprovalRequest.status = this.ApprovalStatus.APPROVED;
+        approvedApprovalRequest.memo = memo;
+
+        this.approvalRequestDAO.put(approvedApprovalRequest).then((req) => {
+          this.approvalRequestDAO.cmd(this.AbstractDAO.RESET_CMD);
+          this.finished.pub();
+          this.notify(this.SUCCESS_APPROVED, '', this.LogLevel.INFO, true);
+
+          if ( this.subStack.top.length > 0 ) {
+            this.subStack.back();
+          }
+        }, (e) => {
+          this.throwError.pub(e);
+          this.notify(e.message, '', this.LogLevel.ERROR, true);
+        });
+      }
+    },
+    {
+      name: 'rejectWithMemo',
+      code: function(memo) {
+        var rejectedApprovalRequest = this.clone();
+        rejectedApprovalRequest.status = this.ApprovalStatus.REJECTED;
+        rejectedApprovalRequest.memo = memo;
+
+        this.approvalRequestDAO.put(rejectedApprovalRequest).then((o) => {
+          this.approvalRequestDAO.cmd(this.AbstractDAO.RESET_CMD);
+          this.finished.pub();
+          this.notify(this.SUCCESS_REJECTED, '', this.LogLevel.INFO, true);
+
+          if ( this.subStack.top.length > 0 ) {
+            this.subStack.back();
+          }
+        }, (e) => {
+          this.throwError.pub(e);
+          this.notify(e.message, '', this.LogLevel.ERROR, true);
+        });
+      }
+    }
   ]
 });
+
+foam.CLASS({
```

Comment:
Should be in it's own file, or in an inner-class (classes:) in the class/model that instantiates it, which shouldn't be this one.